simple/msg_setname.c

```diff
@@ -0,0 +1,647 @@
+		ret = -FI_EOTHER;
+		goto err6;
+	}
+
+	ret = check_address(&ep->fid, "connect");
+	if (ret) {
+		goto err6;
+	}
+
+	fi_freeinfo(fi);
+	return 0;
+
+err6:
+	free_ep_res();
+err5:
+	fi_close(&cmeq->fid);
+err4:
+	fi_close(&dom->fid);
+err2:
+	fi_freeinfo(fi);
+err0:
+	return ret;
+}
+
+static int send_recv()
+{
+	struct fi_cq_entry comp;
+	int ret;
+
+	if (opts.dst_addr) {
+		/* Client */
+		fprintf(stdout, "Posting a send...\n");
+		sprintf(buf, "Hello World!"); 
+		ret = fi_send(ep, buf, sizeof("Hello World!"), fi_mr_desc(mr), 0, buf);
+		if (ret) {
+			FT_PRINTERR("fi_send", ret);
+			return ret;
+		}
+
+		/* Read send queue */
+		do {
+			ret = fi_cq_read(scq, &comp, 1);
+			if (ret < 0 && ret != -FI_EAGAIN) {
+				FT_PRINTERR("fi_cq_read", ret);
+				return ret;
+			}
+		} while (ret == -FI_EAGAIN);
+
+		fprintf(stdout, "Send completion received\n");
+	} else {
+		/* Server */
+		fprintf(stdout, "Posting a recv...\n");
+		ret = fi_recv(ep, buf, buffer_size, fi_mr_desc(mr), 0, buf);
+		if (ret) {
+			FT_PRINTERR("fi_recv", ret);
+			return ret;
+		}
+
+		/* Read recv queue */
+		fprintf(stdout, "Waiting for client...\n");
+		do {
+			ret = fi_cq_read(rcq, &comp, 1);
+			if (ret < 0 && ret != -FI_EAGAIN) {
+				FT_PRINTERR("fi_cq_read", ret);
+				return ret;
+			}
+		} while (ret == -FI_EAGAIN);
+
+		fprintf(stdout, "Received data from client: %s\n", (char *)buf);
+	}
+
+	return 0;
+}
+
+static int setup_handle(void)
+{
+	static char buf[BUFSIZ];
+	struct addrinfo *ai, aihints;
+	struct fi_info *fi;
+	int ret;
+
+	memset(&aihints, 0, sizeof aihints);
+	aihints.ai_flags = AI_PASSIVE;
+	ret = getaddrinfo(opts.src_addr, opts.src_port, &aihints, &ai);
+	if (ret == EAI_SYSTEM) {
+		FT_PRINTERR("getaddrinfo", -ret);
+		return -ret;
+	} else if (ret) {
+		FT_ERR("getaddrinfo: %s\n", gai_strerror(ret));
+		return -FI_ENODATA;
+	}
+
+	switch (ai->ai_family) {
+	case AF_INET:
+		hints->addr_format = FI_SOCKADDR_IN;
+		break;
+	case AF_INET6:
+		hints->addr_format = FI_SOCKADDR_IN6;
+		break;
+	case AF_IB:
+		hints->addr_format = FI_SOCKADDR_IB;
+		break;
+	}
+
+	/* Get fabric info */
+	ret = fi_getinfo(FT_FIVERSION, NULL, NULL, 0, hints, &fi);
+	if (ret) {
+		FT_PRINTERR("fi_getinfo", ret);
+		goto free_ai;
+	}
+
+	/* Open the fabric */
+	ret = fi_fabric(fi->fabric_attr, &fab, NULL);
+	if (ret) {
+		FT_PRINTERR("fi_fabric", ret);
+		goto free_fi;
+	}
+
+	/* Open a passive endpoint */
+	ret = fi_passive_ep(fab, fi, &pep, NULL);
+	if (ret) {
+		FT_PRINTERR("fi_passive_ep", ret);
+		goto free_fabric;
+	}
+
+	ret = fi_setname(&pep->fid, ai->ai_addr, ai->ai_addrlen);
+	if (ret) {
+		FT_PRINTERR("fi_setname", ret);
+		goto free_pep;
+	}
+
+	ret = fi_getname(&pep->fid, bound_addr, &bound_addr_len);
+	if (ret) {
+		FT_PRINTERR("fi_getname", ret);
+		goto free_pep;
+	} else if (bound_addr->sa_family != AF_INET) {
+		FT_ERR("got unexpected address family %d\n",
+				bound_addr->sa_family);
```

Comment:
_Trying this reply again since the Android e-mail client apparently strips out all headers on reply :(_

That was a debug check that was accidentally left in.  Will fix.